python/lsst/daf/butler/core/datasets.py

```diff
@@ -227,6 +227,30 @@ def __deepcopy__(self, memo):
                            dimensions=deepcopy(self.dimensions, memo),
                            storageClass=deepcopy(self._storageClass or self._storageClassName, memo))
 
+    def normalize(self, universe):
+        """Ensure the dimensions and storage class name are valid, and make
+        ``self.dimensions`` a true `DimensionGraph` instance if it isn't
+        already.
+
+        Parameters
+        ----------
+        universe : `DimensionGraph`
+            The set of all known dimensions.
+
+        Raises
+        ------
+        ValueError
+            Raised if the DatasetType is invalid, either because one or more
+            dimensions in ``self.dimensions`` is not in ``universe``, or the
+            storage class name is not recognized.
+        """
+        if not isinstance(self._dimensions, DimensionGraph):
+            self._dimensions = universe.extract(self._dimensions)
+        try:
+            self.storageClass
```

Comment:
It's because it's looking up the storage class instance from a global registry, given the storage class name; I'm just accessing this property to make it do that.
I'm not sure it's the right exception class, but it's not obviously wrong, either, and I'd rather not change it on this ticket.  I will add a comment explaining what is going on.